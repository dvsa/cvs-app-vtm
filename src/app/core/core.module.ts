import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
