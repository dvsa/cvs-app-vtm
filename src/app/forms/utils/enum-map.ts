import { MultiOptions } from '@forms/models/options.model';

export function getOptionsFromEnum(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() + value.slice(1) }));
}

export function getOptionsFromEnumOneChar(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.charAt(0).toUpperCase() }));
}

export function getOptionsFromEnumAcronym(object: object): MultiOptions {
  return Object.values(object).map(value => ({ value, label: value.toUpperCase() }));
}
