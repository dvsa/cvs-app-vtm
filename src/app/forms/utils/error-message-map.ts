import { formatDate } from '@angular/common';
import { AsyncValidatorNames } from '@forms/models/async-validators.enum';
import { ValidatorNames } from '@forms/models/validators.enum';

const DEFAULT_LABEL = 'This field';
export const ErrorMessageMap: Record<string, (...args: any) => string> = {
  // Date errors
  invalidDate: (err: { error: boolean; reason: string; index: number }) => `${err.reason}`,
  invalidOption: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is invalid`,
  invalidTestResult: (err: { message: string }) => err.message,

  [ValidatorNames.AheadOfDate]: (err: { sibling: string; date: Date }, label?: string) =>
    `${label || 'This date'} must be ahead of ${err.sibling || 'the previous date'}${err.date ? formatDate(err.date, ' (dd/MM/yyyy)', 'en') : ''}`,
  [ValidatorNames.CustomPattern]: (err: { message: string }, label?: string) => `${label || DEFAULT_LABEL} ${err.message}`,
  [ValidatorNames.DateNotExceed]: (err: { sibling: string; months: number }, label?: string) =>
    `${label || 'This date'} must be less than ${err.months || 'X'} months after ${err.sibling || 'the previous date'}`,
  [ValidatorNames.Defined]: (err: boolean, label?: string) => `${label} is required`,
  [ValidatorNames.FutureDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the future`,
  [ValidatorNames.Max]: (err: { max: number }, label?: string) => `${label || DEFAULT_LABEL} must be less than or equal to ${err.max}`,
  [ValidatorNames.MaxLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be less than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.Min]: (err: { min: number }, label?: string) => `${label || DEFAULT_LABEL} must be greater than or equal to ${err.min}`,
  [ValidatorNames.MinLength]: (err: { requiredLength: number }, label?: string) =>
    `${label || DEFAULT_LABEL} must be greater than or equal to ${err.requiredLength} characters`,
  [ValidatorNames.NotZNumber]: () => "The VRM/Trailer ID cannot be in a format that is 7 digits followed by the character 'Z'",
  [ValidatorNames.PastDate]: (err: boolean, label?: string) => `${label || 'This date'} must be in the past`,
  [ValidatorNames.Pattern]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} must match a pattern`,
  [ValidatorNames.Required]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [ValidatorNames.RequiredIfEquals]: (err: { sibling: string }, label?: string) => `${label || DEFAULT_LABEL} is required with ${err.sibling}`,
  [ValidatorNames.ValidateDefectNotes]: () => 'Notes is required',
  [ValidatorNames.ValidateProhibitionIssued]: () => 'Prohibition notice has not been issued.',

  [AsyncValidatorNames.RequiredIfNotAbandoned]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotFail]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResult]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`,
  [AsyncValidatorNames.RequiredIfNotResultAndSiblingEquals]: (err: boolean, label?: string) => `${label || DEFAULT_LABEL} is required`
};
