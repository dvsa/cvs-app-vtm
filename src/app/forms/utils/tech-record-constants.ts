import { FormNode } from '@forms/services/dynamic-form.types';
import { CarTechRecord } from '@forms/templates/car/car-tech-record.template';
import { ApplicantDetails } from '@forms/templates/general/applicant-details.template';
import { HgvAndTrlTypeApprovalTemplate } from '@forms/templates/general/approval-type.template';
import { Audit } from '@forms/templates/general/audit.template';
import { DocumentsTemplate } from '@forms/templates/general/documents.template';
import { HgvAndTrlBodyTemplate } from '@forms/templates/general/hgv-trl-body.template';
import { LettersTemplate } from '@forms/templates/general/letters.template';
import { ManufacturerTemplate } from '@forms/templates/general/manufacturer.template';
import { NotesTemplate } from '@forms/templates/general/notes.template';
import { PlatesTemplate } from '@forms/templates/general/plates.template';
import { HgvDimensionsTemplate } from '@forms/templates/hgv/hgv-dimensions.template';
import { HgvTechRecord } from '@forms/templates/hgv/hgv-tech-record.template';
import { tyresTemplateHgv } from '@forms/templates/hgv/hgv-tyres.template';
import { HgvWeight } from '@forms/templates/hgv/hgv-weight.template';
import { LgvTechRecord } from '@forms/templates/lgv/lgv-tech-record.template';
import { MotorcycleTechRecord } from '@forms/templates/motorcycle/motorcycle-tech-record.template';
import { PsvTypeApprovalTemplate } from '@forms/templates/psv/psv-approval-type.template';
import { PsvBodyTemplate } from '@forms/templates/psv/psv-body.template';
import { PsvBrakesTemplate } from '@forms/templates/psv/psv-brakes.template';
import { PsvDdaTemplate } from '@forms/templates/psv/psv-dda.template';
import { PsvDimensionsTemplate } from '@forms/templates/psv/psv-dimensions.template';
import { PsvNotes } from '@forms/templates/psv/psv-notes.template';
import { PsvTechRecord } from '@forms/templates/psv/psv-tech-record.template';
import { PsvTyresTemplate } from '@forms/templates/psv/psv-tyres.template';
import { PsvWeightsTemplate } from '@forms/templates/psv/psv-weight.template';
import { SmallTrailerTechRecord } from '@forms/templates/small-trailer/small-trailer-tech-record.template';

import { TechRecordReasonForCreationSection } from '@forms/templates/test-records/section-templates/techRecord_reasonForCreation/techRecordReasonForCreation.template';
import { TrlAuthIntoServiceTemplate } from '@forms/templates/trl/trl-auth-into-service.template';
import { TrlBrakesTemplate } from '@forms/templates/trl/trl-brakes.template';
import { TrlDimensionsTemplate } from '@forms/templates/trl/trl-dimensions.template';
import { TrlPurchasers } from '@forms/templates/trl/trl-purchaser.template';
import { TrlTechRecordTemplate } from '@forms/templates/trl/trl-tech-record.template';
import { tyresTemplateTrl } from '@forms/templates/trl/trl-tyres.template';
import { TrlWeight } from '@forms/templates/trl/trl-weight.template';
import { VehicleTypes } from '@models/vehicle-tech-record.model';

// The map below initializes the array of sections that the *ngFor in tech summary component's template will iterate over.
// The order in which each section is introduced in the array will determine its order on the page when rendered.
// Sections which use custom components require a FormNode object with 'name' and 'label' properties.

export const vehicleTemplateMap = new Map<VehicleTypes, Array<FormNode>>([
  [
    VehicleTypes.PSV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ PsvNotes,
      /*  3 */ PsvTechRecord,
      /*  4 */ PsvTypeApprovalTemplate,
      /*  5 */ PsvBrakesTemplate,
      /*  6 */ PsvDdaTemplate,
      /*  7 */ DocumentsTemplate,
      /*  8 */ PsvBodyTemplate,
      /*  9 */ PsvWeightsTemplate,
      /* 10 */ PsvTyresTemplate,
      /* 11 */ PsvDimensionsTemplate
    ]
  ],
  [
    VehicleTypes.HGV,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ HgvTechRecord,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ HgvAndTrlBodyTemplate,
      /*  8 */ HgvWeight,
      /*  9 */ tyresTemplateHgv,
      /* 10 */ HgvDimensionsTemplate,
      /* 11 */ PlatesTemplate
    ]
  ],
  [
    VehicleTypes.TRL,
    [
      /*  1 */ TechRecordReasonForCreationSection,
      /*  2 */ NotesTemplate,
      /*  3 */ TrlTechRecordTemplate,
      /*  4 */ HgvAndTrlTypeApprovalTemplate,
      /*  5 */ ApplicantDetails,
      /*  6 */ DocumentsTemplate,
      /*  7 */ LettersTemplate,
      /*  8 */ HgvAndTrlBodyTemplate,
      /*  9 */ TrlWeight,
      /* 10 */ tyresTemplateTrl,
      /* 11 */ TrlBrakesTemplate,
      /* 12 */ TrlPurchasers,
      /* 13 */ TrlDimensionsTemplate,
      /* 14 */ PlatesTemplate,
      /* 15 */ TrlAuthIntoServiceTemplate,
      /* 16 */ ManufacturerTemplate
    ]
  ],
  [
    VehicleTypes.SMALL_TRL,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ SmallTrailerTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.LGV,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ LgvTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.CAR,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ CarTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ],
  [
    VehicleTypes.MOTORCYCLE,
    [/* 1 */ TechRecordReasonForCreationSection, /* 2 */ MotorcycleTechRecord, /* 3 */ ApplicantDetails, /* 4 */ NotesTemplate, /* 5 */ Audit]
  ]
]);
