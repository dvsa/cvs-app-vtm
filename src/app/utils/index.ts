export * from './mockStore';
export * from './testing.utils';
