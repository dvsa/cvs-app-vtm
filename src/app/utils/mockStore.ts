import { map } from 'rxjs/operators';
import { BehaviorSubject } from 'rxjs';
import { Action } from '@ngrx/store';
import { getErrors } from '@app/store/selectors/error.selectors';
import {
  getSelectedVehicleTechRecord,
  getTechViewState,
  getVehicleTechRecordMetaData,
  getActiveVehicleTechRecord,
  selectVehicleTechRecordModelHavingStatusAll
} from '@app/store/selectors/VehicleTechRecordModel.selectors';
import {
  getSelectedVehicleTestResultModel,
  getTestViewState,
  getVehicleTestResultModel,
  selectTestTypeById
} from '@app/store/selectors/VehicleTestResultModel.selectors';
import { getPreparers, getTestStations } from '@app/store/selectors/ReferenceData.selectors';
import { getRouterParams } from '@app/store/selectors/route.selectors';

export class MockStore {
  mockSelector: BehaviorSubject<any>;

  constructor(mockSelector) {
    this.mockSelector = mockSelector;
  }

  select(selector: any) {
    switch (selector) {
      case selectVehicleTechRecordModelHavingStatusAll:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectVehicleTechRecordModelHavingStatusAll')
              ? value['selectVehicleTechRecordModelHavingStatusAll']
              : []
          )
        );

      case getSelectedVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTechRecord')
              ? value['getSelectedVehicleTechRecord']
              : {}
          )
        );

      case getActiveVehicleTechRecord:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getActiveVehicleTechRecord')
              ? value['getActiveVehicleTechRecord']
              : () => {}
          )
        );

      case getVehicleTechRecordMetaData:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTechRecordMetaData')
              ? value['getVehicleTechRecordMetaData']
              : {}
          )
        );

      case getTechViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTechViewState') ? value['getTechViewState'] : {}
          )
        );

      case getErrors:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getErrors') ? value['getErrors'] : []
          )
        );

      case getVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getVehicleTestResultModel')
              ? value['getVehicleTestResultModel']
              : []
          )
        );

      case selectTestTypeById(''):
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('selectTestTypeById') ? value['selectTestTypeById'] : {}
          )
        );

      case getPreparers:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getPreparers') ? value['getPreparers'] : null
          )
        );

      case getTestStations:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestStations') ? value['getTestStations'] : null
          )
        );

      case getTestViewState:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getTestViewState') ? value['getTestViewState'] : {}
          )
        );

      case getSelectedVehicleTestResultModel:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getSelectedVehicleTestResultModel')
              ? value['getSelectedVehicleTestResultModel']
              : {}
          )
        );

      case getRouterParams:
        return this.mockSelector.pipe(
          map((value: any) =>
            value && value.hasOwnProperty('getRouterParams') ? value['getRouterParams'] : {}
          )
        );

      default:
        return this.mockSelector;
    }
  }

  dispatch(action: Action) {
    return [];
  }
}
