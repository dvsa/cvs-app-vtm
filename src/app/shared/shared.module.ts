import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { DocumentRetrievalService } from '@api/document-retrieval';
import { RoleRequiredDirective } from '@directives/app-role-required.directive';
import { AccordionControlComponent } from './components/accordion-control/accordion-control.component';
import { AccordionComponent } from './components/accordion/accordion.component';
import { BannerComponent } from './components/banner/banner.component';
import { BaseDialogComponent } from './components/base-dialog/base-dialog.component';
import { ButtonGroupComponent } from './components/button-group/button-group.component';
import { ButtonComponent } from './components/button/button.component';
import { IconComponent } from './components/icon/icon.component';
import { NumberPlateComponent } from './components/number-plate/number-plate.component';
import { PaginationComponent } from './components/pagination/pagination.component';
import { TagComponent } from './components/tag/tag.component';
import { TestCertificateComponent } from './components/test-certificate/test-certificate.component';
import { RetrieveDocumentDirective } from './directives/retrieve-document/retrieve-document.directive';
import { PreventDoubleClickDirective } from './directives/prevent-double-click/prevent-double-click.directive';
import { DefaultNullOrEmpty } from './pipes/default-null-or-empty/default-null-or-empty.pipe';
import { DigitGroupSeparatorPipe } from './pipes/digit-group-separator/digit-group-separator.pipe';
import { RefDataDecodePipe } from './pipes/ref-data-decode/ref-data-decode.pipe';
import { TestTypeNamePipe } from './pipes/test-type-name/test-type-name.pipe';

@NgModule({
  declarations: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    PreventDoubleClickDirective,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  imports: [CommonModule, RouterModule],
  exports: [
    DefaultNullOrEmpty,
    ButtonGroupComponent,
    ButtonComponent,
    BannerComponent,
    RoleRequiredDirective,
    TagComponent,
    NumberPlateComponent,
    IconComponent,
    TestTypeNamePipe,
    AccordionComponent,
    AccordionControlComponent,
    PaginationComponent,
    TestCertificateComponent,
    BaseDialogComponent,
    DigitGroupSeparatorPipe,
    RefDataDecodePipe,
    RetrieveDocumentDirective
  ],
  providers: [DocumentRetrievalService]
})
export class SharedModule {}
