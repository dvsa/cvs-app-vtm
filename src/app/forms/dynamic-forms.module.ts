import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NoSpaceDirective } from '@directives/app-no-space/app-no-space.directive';
import { NumberOnlyDirective } from '@directives/app-number-only/app-number-only.directive';
import { ToUppercaseDirective } from '@directives/app-to-uppercase/app-to-uppercase.directive';
import { TrimWhitespaceDirective } from '@directives/app-trim-whitespace/app-trim-whitespace.directive';
import { ApprovalTypeFocusNextDirective } from '@directives/approval-type-focus-next/approval-type-focus-next.directive';
import { DateFocusNextDirective } from '@directives/date-focus-next/date-focus-next.directive';
import { PrefixDirective } from '@directives/prefix/prefix.directive';
import { SuffixDirective } from '@directives/suffix/suffix.directive';
import { ApprovalTypeInputComponent } from '@forms/components/approval-type/approval-type.component';
import { AdrCertificateHistoryComponent } from '@forms/custom-sections/adr-certificate-history/adr-certificate-history.component';
import { AdrExaminerNotesHistoryEditComponent } from '@forms/custom-sections/adr-examiner-notes-history-edit/adr-examiner-notes-history.component-edit';
import { AdrPermittedDangerousGoodsComponent } from '@forms/custom-sections/adr-permitted-dangerous-goods/adr-permitted-dangerous-goods.component';
import { ApprovalTypeComponent } from '@forms/custom-sections/approval-type/approval-type.component';
import { SharedModule } from '@shared/shared.module';
import { TruncatePipe } from '../pipes/truncate/truncate.pipe';
import { AutocompleteComponent } from './components/autocomplete/autocomplete.component';
import { BaseControlComponent } from './components/base-control/base-control.component';
import { CharacterCountComponent } from './components/character-count/character-count.component';
import { CheckboxGroupComponent } from './components/checkbox-group/checkbox-group.component';
import { CheckboxComponent } from './components/checkbox/checkbox.component';
import { ContingencyAdrGenerateCertComponent } from './components/contingency-adr-generate-cert/contingency-adr-generate-cert.component';
import { ControlErrorsComponent } from './components/control-errors/control-errors.component';
import { DateControlsComponent } from './components/date-controls/date-controls.component';
import { DateComponent } from './components/date/date.component';
import { DefectSelectComponent } from './components/defect-select/defect-select.component';
import { DynamicFormFieldComponent } from './components/dynamic-form-field/dynamic-form-field.component';
import { DynamicFormGroupComponent } from './components/dynamic-form-group/dynamic-form-group.component';
import { FieldErrorMessageComponent } from './components/field-error-message/field-error-message.component';
import { FieldWarningMessageComponent } from './components/field-warning-message/field-warning-message.component';
import { NumberInputComponent } from './components/number-input/number-input.component';
import { RadioGroupComponent } from './components/radio-group/radio-group.component';
import { ReadOnlyComponent } from './components/read-only/read-only.component';
import { RequiredStandardSelectComponent } from './components/required-standard-select/required-standard-select.component';
import { SelectComponent } from './components/select/select.component';
import { SuggestiveInputComponent } from './components/suggestive-input/suggestive-input.component';
import { SwitchableInputComponent } from './components/switchable-input/switchable-input.component';
import { TextAreaComponent } from './components/text-area/text-area.component';
import { TextInputComponent } from './components/text-input/text-input.component';
import { ViewCombinationComponent } from './components/view-combination/view-combination.component';
import { ViewListItemComponent } from './components/view-list-item/view-list-item.component';
import { AbandonDialogComponent } from './custom-sections/abandon-dialog/abandon-dialog.component';
import { AdrExaminerNotesHistoryViewComponent } from './custom-sections/adr-examiner-notes-history-view/adr-examiner-notes-history-view.component';
import { AdrNewCertificateRequiredViewComponent } from './custom-sections/adr-new-certificate-required-view/adr-new-certificate-required-view.component';
import { AdrSectionEditComponent } from './custom-sections/adr-section/adr-section-edit/adr-section-edit.component';
import { AdrSectionSummaryComponent } from './custom-sections/adr-section/adr-section-summary/adr-section-summary.component';
import { AdrSectionViewComponent } from './custom-sections/adr-section/adr-section-view/adr-section-view.component';
import { AdrSectionComponent } from './custom-sections/adr-section/adr-section.component';
import { AdrTankDetailsInitialInspectionViewComponent } from './custom-sections/adr-tank-details-initial-inspection-view/adr-tank-details-initial-inspection-view.component';
import { AdrTankDetailsM145ViewComponent } from './custom-sections/adr-tank-details-m145-view/adr-tank-details-m145-view.component';
import { AdrTankDetailsSubsequentInspectionsEditComponent } from './custom-sections/adr-tank-details-subsequent-inspections-edit/adr-tank-details-subsequent-inspections-edit.component';
import { AdrTankDetailsSubsequentInspectionsViewComponent } from './custom-sections/adr-tank-details-subsequent-inspections-view/adr-tank-details-subsequent-inspections-view.component';
import { AdrTankStatementUnNumberEditComponent } from './custom-sections/adr-tank-statement-un-number-edit/adr-tank-statement-un-number-edit.component';
import { AdrTankStatementUnNumberViewComponent } from './custom-sections/adr-tank-statement-un-number-view/adr-tank-statement-un-number-view.component';
import { AdrComponent } from './custom-sections/adr/adr.component';
import { BodyComponent } from './custom-sections/body/body.component';
import { CustomDefectComponent } from './custom-sections/custom-defect/custom-defect.component';
import { CustomDefectsComponent } from './custom-sections/custom-defects/custom-defects.component';
import { CustomFormControlComponent } from './custom-sections/custom-form-control/custom-form-control.component';
import { DefectComponent } from './custom-sections/defect/defect.component';
import { DefectsComponent } from './custom-sections/defects/defects.component';
import { DimensionsComponent } from './custom-sections/dimensions/dimensions.component';
import { LettersComponent } from './custom-sections/letters/letters.component';
import { ModifiedWeightsComponent } from './custom-sections/modified-weights/modified-weights.component';
import { PlatesComponent } from './custom-sections/plates/plates.component';
import { PsvBrakesComponent } from './custom-sections/psv-brakes/psv-brakes.component';
import { RequiredStandardComponent } from './custom-sections/required-standard/required-standard.component';
import { RequiredStandardsComponent } from './custom-sections/required-standards/required-standards.component';
import { TrlBrakesComponent } from './custom-sections/trl-brakes/trl-brakes.component';
import { TyresComponent } from './custom-sections/tyres/tyres.component';
import { WeightsComponent } from './custom-sections/weights/weights.component';

@NgModule({
	declarations: [
		BaseControlComponent,
		TextInputComponent,
		ViewListItemComponent,
		DynamicFormGroupComponent,
		ViewCombinationComponent,
		CheckboxGroupComponent,
		RadioGroupComponent,
		DefectComponent,
		DefectsComponent,
		AutocompleteComponent,
		NumberInputComponent,
		TextAreaComponent,
		NumberOnlyDirective,
		ToUppercaseDirective,
		NoSpaceDirective,
		TrimWhitespaceDirective,
		DateComponent,
		DateControlsComponent,
		CharacterCountComponent,
		ControlErrorsComponent,
		SelectComponent,
		DynamicFormFieldComponent,
		FieldErrorMessageComponent,
		DefectSelectComponent,
		RequiredStandardSelectComponent,
		DateFocusNextDirective,
		TruncatePipe,
		WeightsComponent,
		LettersComponent,
		PlatesComponent,
		DimensionsComponent,
		TrlBrakesComponent,
		ReadOnlyComponent,
		CustomDefectsComponent,
		RequiredStandardComponent,
		RequiredStandardsComponent,
		CustomDefectComponent,
		SwitchableInputComponent,
		ReadOnlyComponent,
		SuffixDirective,
		AbandonDialogComponent,
		BodyComponent,
		TyresComponent,
		PsvBrakesComponent,
		PrefixDirective,
		SuggestiveInputComponent,
		CheckboxComponent,
		ApprovalTypeComponent,
		ApprovalTypeInputComponent,
		ApprovalTypeFocusNextDirective,
		ModifiedWeightsComponent,
		FieldWarningMessageComponent,
		AdrComponent,
		AdrTankDetailsSubsequentInspectionsEditComponent,
		AdrTankStatementUnNumberEditComponent,
		CustomFormControlComponent,
		AdrExaminerNotesHistoryEditComponent,
		AdrExaminerNotesHistoryViewComponent,
		AdrTankDetailsSubsequentInspectionsViewComponent,
		AdrTankDetailsInitialInspectionViewComponent,
		AdrTankStatementUnNumberViewComponent,
		AdrCertificateHistoryComponent,
		AdrTankDetailsM145ViewComponent,
		ContingencyAdrGenerateCertComponent,
		AdrNewCertificateRequiredViewComponent,
		AdrSectionComponent,
		AdrSectionEditComponent,
		AdrSectionViewComponent,
		AdrSectionSummaryComponent,
		AdrPermittedDangerousGoodsComponent,
	],
	imports: [CommonModule, FormsModule, ReactiveFormsModule, SharedModule, RouterModule],
	exports: [
		TextInputComponent,
		ViewListItemComponent,
		DynamicFormGroupComponent,
		ViewCombinationComponent,
		CheckboxGroupComponent,
		RadioGroupComponent,
		DefectComponent,
		DefectsComponent,
		AutocompleteComponent,
		NumberInputComponent,
		TextAreaComponent,
		DateComponent,
		DateControlsComponent,
		CharacterCountComponent,
		ControlErrorsComponent,
		SelectComponent,
		DynamicFormFieldComponent,
		FieldErrorMessageComponent,
		DefectSelectComponent,
		RequiredStandardSelectComponent,
		WeightsComponent,
		LettersComponent,
		PlatesComponent,
		TyresComponent,
		DimensionsComponent,
		TrlBrakesComponent,
		ReadOnlyComponent,
		RequiredStandardComponent,
		RequiredStandardsComponent,
		CustomDefectsComponent,
		CustomDefectComponent,
		SwitchableInputComponent,
		SuffixDirective,
		ReadOnlyComponent,
		AbandonDialogComponent,
		BodyComponent,
		PsvBrakesComponent,
		PrefixDirective,
		SuggestiveInputComponent,
		CheckboxComponent,
		ToUppercaseDirective,
		NoSpaceDirective,
		TrimWhitespaceDirective,
		ApprovalTypeComponent,
		ApprovalTypeInputComponent,
		ApprovalTypeFocusNextDirective,
		ModifiedWeightsComponent,
		AdrComponent,
		AdrCertificateHistoryComponent,
		FieldWarningMessageComponent,
		AdrSectionComponent,
		AdrSectionEditComponent,
		AdrSectionViewComponent,
		AdrSectionSummaryComponent,
	],
})
export class DynamicFormsModule {}
