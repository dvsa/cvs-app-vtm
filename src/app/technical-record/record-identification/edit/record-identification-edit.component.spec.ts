import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule, FormsModule, FormGroupDirective } from '@angular/forms';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { RecordIdentificationEditComponent } from './record-identification-edit.component';

describe('RecordIdentificationEditComponent', () => {
  let component: RecordIdentificationEditComponent;
  let fixture: ComponentFixture<RecordIdentificationEditComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule, SharedModule],
      declarations: [RecordIdentificationEditComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      providers: [
        FormGroupDirective,
        {
          provide: FormGroupDirective,
          useValue: TESTING_UTILS.mockFormGroupDirective()
        }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RecordIdentificationEditComponent);
    component = fixture.componentInstance;
  });

  it('should create with initialized form controls', () => {
    component.vin = 'ABCDEFGH654321';
    component.primaryVrm = 'LKJH654';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
