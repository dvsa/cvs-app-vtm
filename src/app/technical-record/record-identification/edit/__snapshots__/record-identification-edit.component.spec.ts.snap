// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordIdentificationEditComponent should create with initialized form controls 1`] = `
<vtm-record-identification-edit
  fb={[Function FormBuilder]}
  parent={[Function FormGroupDirective]}
  primaryVrm={[Function String]}
  vehicleRecordFg={[Function FormGroup]}
  vin={[Function String]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-three-quarters"
    >
      <div
        class="govuk-form-group"
      >
        <fieldset
          class="govuk-fieldset"
        >
          <form
            class="ng-untouched ng-pristine ng-valid"
            ng-reflect-form="[object Object]"
            novalidate=""
          >
            <div
              class="govuk-form-group govuk-!-margin-top-7"
            >
              <div
                class="govuk-form-group"
              >
                <label
                  class="govuk-label govuk-!-font-weight-bold"
                  for="vin"
                >
                   Vehicle identification number (VIN) 
                </label>
                <input
                  class="govuk-input govuk-input--width-20 ng-untouched ng-pristine ng-valid"
                  formcontrolname="vin"
                  id="vin"
                  ng-reflect-name="vin"
                  type="text"
                />
              </div>
              <div
                class="govuk-form-group"
              >
                <label
                  class="govuk-label govuk-!-font-weight-bold"
                  for="vrm"
                >
                   Vehicle registration number (VRM) 
                </label>
                <input
                  class="govuk-input govuk-input--width-10 ng-untouched ng-pristine ng-valid"
                  formcontrolname="primaryVrm"
                  id="vrm"
                  ng-reflect-name="primaryVrm"
                  type="text"
                />
              </div>
            </div>
          </form>
        </fieldset>
      </div>
    </div>
  </div>
</vtm-record-identification-edit>
`;
